LEGO1/tgl/d3drm/mesh.cpp

```diff
@@ -65,43 +65,52 @@ Result MeshImpl::SetTexture(const Texture* pTexture)
 
 // FUNCTION: LEGO1 0x100a3f80
 // FUNCTION: BETA10 0x10170690
-Result MeshImpl::SetTextureMappingMode(TextureMappingMode mode)
+Tgl::Result MeshImpl::SetTextureMappingMode(TextureMappingMode mode)
 {
 	assert(m_data);
 
 	return MeshSetTextureMappingMode(m_data, mode);
 }
 
 // FUNCTION: BETA10 0x10170750
-inline Result MeshSetShadingModel(MeshImpl::MeshData* pMesh, ShadingModel model)
+inline Tgl::Result MeshSetShadingModel(MeshImpl::MeshData* pMesh, ShadingModel model)
 {
 	D3DRMRENDERQUALITY mode = Translate(model);
 	return ResultVal(pMesh->groupMesh->SetGroupQuality(pMesh->groupIndex, mode));
 }
 
 // FUNCTION: LEGO1 0x100a3fc0
 // FUNCTION: BETA10 0x101706f0
-Result MeshImpl::SetShadingModel(ShadingModel model)
+Tgl::Result MeshImpl::SetShadingModel(ShadingModel model)
 {
 	assert(m_data);
 	return MeshSetShadingModel(m_data, model);
 }
 
 // FUNCTION: BETA10 0x101714e0
-inline Result MeshDeepClone(MeshImpl::MeshData* pSource, MeshImpl::MeshData*& rpTarget, IDirect3DRMMesh* pMesh)
+inline Tgl::Result MeshDeepClone(MeshImpl::MeshData* pSource, MeshImpl::MeshData*& rpTarget, IDirect3DRMMesh* pMesh)
 {
 	rpTarget = new MeshImpl::MeshData();
 	rpTarget->groupMesh = pMesh;
 
 	// Query information from old group
 	DWORD dataSize;
+#if defined(__3DS__)
+	unsigned long vcount, fcount, vperface;
```

Comment:
This does not appear to be needed?

Reply:
Annoyingly the compiler errors with some typing. What I applied was again a "silly patch" but this was just done for the sake of getting it to compile.

It probably is not needed but we'll see

Reply:
maybe you can provide the error